diff --git a/dist/addons/karpenter/index.js b/dist/addons/karpenter/index.js
index 88242d52de482707a1f0813264d1b09db69f25c5..5be655072d1e8ee2a019147d8123a89ecf0fb5df 100644
--- a/dist/addons/karpenter/index.js
+++ b/dist/addons/karpenter/index.js
@@ -143,12 +143,12 @@ class KarpenterAddOn extends helm_addon_1.HelmAddOn {
         sa.node.addDependency(ns);
         // Add helm chart
         if (semver.gte(this.options.version, '0.19.0')) {
-            (0, utils_1.setPath)(values, "settings.aws", {
+            (0, utils_1.setPath)(values, "settings.aws", (0, ts_deepmerge_1.default)({
                 clusterEndpoint: endpoint,
                 clusterName: name,
                 defaultInstanceProfile: karpenterInstanceProfile.instanceProfileName,
                 interruptionQueueName: stackName
-            });
+            }, values?.settings?.aws ?? {}));
         }
         else {
             (0, utils_1.setPath)(values, "clusterEndpoint", endpoint);